pkg/apis/mongodb/v1/mongodb_types.go

```diff
@@ -68,13 +68,24 @@ type MongoDBSpec struct {
 	AdditionalMongodConfig MongodConfiguration `json:"additionalMongodConfig,omitempty"`
 }
 
-func (m MongoDBSpec) ToJson() []byte {
+func (m MongoDBSpec) ToJson() ([]byte, error) {
 	b, err := json.Marshal(m)
 	if err != nil {
-		return []byte{}
+		return []byte{}, err
 	}
 
-	return b
+	return b, nil
+}
+
+func FromJSON(jsonString string) (MongoDBSpec, error) {
+	m := MongoDBSpec{}
+	err := json.Unmarshal([]byte(jsonString), &m)
+	if err != nil {
+		return MongoDBSpec{}, err
+	}
+
+	return m, nil
+
```

Comment:
[q] do these functions actually provide us any value? (applies to the ToJson function aswell)

We are just wrapping one function call, the calling. code will look almost the same anyway

```golang
spec, err := mdbv1.FromJson(jsonString)
if err != nil {
   // Handle error
}

// use spec
```

vs 

```golang
spec := MongoDBSpec{}
err := json.Unmarshal([]byte(jsonString), &spec)
if err != nil {
     // handle error
}
// use spec
```